src/lua/util.rs

```diff
@@ -317,3 +319,29 @@ pub fn to_yaml<'a>(util: Table<'a>, lua: &Lua) -> Result<Table<'a>> {
     util.set("to_yaml", func)?;
     Ok(util)
 }
+
+/// Wrap the given text to fit the specified width
+/// It will try to not split words when possible
+///
+/// Type: function( text:string, widthOrOptions: number|WrapOptions ) -> path:string
+///
+/// Example:
+///
+/// ```lua
+/// xplr.util.wrap("this will be cut off", 11)
+/// -- "this will\nbe cut off"
+/// xplr.util.wrap("this will be cut off", { width = 12, initial_indent = "", subsequent_indent = "    " })
+/// -- "this will be\n    cut off"
+/// ```
+pub fn wrap<'a>(util: Table<'a>, lua: &Lua) -> Result<Table<'a>> {
+    let func = lua.create_function(|lua, (text, options): (String, Value)| {
+        if let Ok(width) = lua.from_value::<usize>(options.clone()) {
+            return Ok(textwrap::wrap(&text, width).join("\n"));
+        }
+
+        let options = lua.from_value::<WrapOptions>(options)?;
+        Ok(textwrap::wrap(&text, options.get_options()).join("\n"))
```

Comment:
Actually, I'd return a list of strings (since we also have DynamicList). table.concat() is way easier than splitting strings.